plane/src/drone/executor.rs

```diff
@@ -211,6 +211,21 @@ impl<R: Runtime> Executor<R> {
                     // else we can deadlock.
                     let Some(manager) = self.backends.get(backend_id) else {
                         tracing::warn!(backend_id = backend_id.as_value(), "Backend not found when handling terminate action (assumed terminated).");
+
+                        self.state_store
```

Comment:
Here we're just assuming the backend is no longer running if the `backend_id` is not in `self.backends`. This isn't a _terrible_ assumption, since the drone is supposed to look for and terminate any backends that were stored in the sqlite table. But there are a few spots where this could break (if a backend doesn't get written to the table before the process dies or if the restarted drone agent was unable to send a termination message to the container). 

If it's possible, I think it would be better if we could determine for sure if a container is no longer running (by inspecting and seeing if docker says it's still running) before we assert that it is `Terminated`. Is that possible?

Reply:
Good point. I took the approach of just _always_ attempting to terminate the backend, and changing the semantics of the runtime to not error if a backend was already terminated or doesn't exist.